integration-cli/docker_api_inspect_test.go

```diff
@@ -133,3 +133,34 @@ func (s *DockerSuite) TestInspectApiImageResponse(c *check.C) {
 	c.Assert(stringutils.InSlice(imageJSON.Tags, "busybox:latest"), check.Equals, true)
 	c.Assert(stringutils.InSlice(imageJSON.Tags, "busybox:mytag"), check.Equals, true)
 }
+
+// #17131, #17139, #17173
+func (s *DockerSuite) TestInspectApiEmptyFieldsInConfigPre121(c *check.C) {
+	out, _ := dockerCmd(c, "run", "-d", "busybox", "true")
+
+	cleanedContainerID := strings.TrimSpace(out)
+
+	cases := []string{"1.19", "1.20"}
+	for _, version := range cases {
+		endpoint := fmt.Sprintf("/v%s/containers/%s/json", version, cleanedContainerID)
+		status, body, err := sockRequest("GET", endpoint, nil)
+		c.Assert(err, check.IsNil)
+		c.Assert(status, check.Equals, http.StatusOK)
+
+		var inspectJSON map[string]interface{}
```

Comment:
Bit simpler code if you use: `var inspectJSON struct{ Config map[string]*json.RawMessage }`